### PR TITLE
Add `New window` option in Dock

### LINKED ISSUE #4863: Add a menu option "Open new window" to launch icon on macOS
Add a menu option "Open new window" to launch icon on MAC OS, which would open a new editor window where user can open, for example a demo project.

### PR DESCRIPTION
**Pull request by AGulev**
New option `New window` for task bar menu added. It opens a new instance of the editor.

Fixes https://github.com/defold/defold/issues/4863

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [x] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [x] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: editor/src/java/com/defold/editor/Start.java
Comment (AGulev) [inline editor/src/java/com/defold/editor/Start.java:204]: Screenshot 2023-07-24 at 09 18 35

File: editor/src/java/com/defold/editor/Start.java
Comment (AGulev) [inline editor/src/java/com/defold/editor/Start.java:162]: we have logger here. I missed it.

File: editor/src/clj/editor/app_view.clj
Comment (AGulev) [inline editor/src/clj/editor/app_view.clj:1488]: I would prefer to reuse java code in Clojure. But I think it's not something we want most of the time. So I just added this comment here to make sure we keep it sync.

**Comment by AGulev** [comment]
Tested on win/mac/linux

**Review by britzl — APPROVED** [review]

**Comment by JCash** [comment]
Do we have race condition for resources between the instances?
E.g.:
1) Opening the same project twice
2) "Global" files such as settings etc.
3) Are editor scripts affected?

**Comment by AGulev** [comment]
@JCash I don't know, but we already can create a new project from `File->Open Project...`

**Comment by JCash** [comment]
Ok, waiting for the first issue then :) 🤷

**Comment by AGulev** [comment]
> Ok, waiting for the first issue then :) 🤷

I expect these things were investigated when the initial functionality was implemented.
I think adding one more place for doing the same, users can already do from another place for a couple of months shouldn't affect anything. But I'll wait for review from somebody from the editor team.

File: editor/src/java/com/defold/editor/Start.java
Comment (matgis) [inline editor/src/java/com/defold/editor/Start.java:195]: I think it should be "New Window", with a capitalized "W", to match the capitalization of the other entries in the menu?

File: editor/src/java/com/defold/editor/Start.java
Comment (matgis) [inline editor/src/java/com/defold/editor/Start.java:210]: Can this even happen after the checks we do? If so, it should probably be logged as info, not an error.

File: editor/src/java/com/defold/editor/Start.java
Comment (matgis) [inline editor/src/java/com/defold/editor/Start.java:200]: We shouldn't re-implement the functionality in Java, rather, we should either invoke the handler directly (although I can't find a convenient way to do that? I think is should be done how we do it from `editor.keymap/execute-command`).

Or we could break out the code in the `:open-project` `defhandler` into a Clojure function that we place next to the handler and call the Clojure function from here. Although to be honest I don't see why this code has to be in Java at all. Couldn't we do this from `boot-open-project/load-stage!` where we do all the other Window and Menu setup?

**Review by matgis — CHANGES_REQUESTED** [review]
Unless there is some reason for needing to do this before we do anything else, I think we should move the code that adds the context menu item into `boot-open-project/load-stage!` where we do the rest of the Window and Menu setup. That also makes it easy to invoke the existing Clojure code.

**Pull request closed without merging**
